numpy/lib/type_check.py

```diff
@@ -134,13 +135,17 @@ def real(val):
     >>> a.real = np.array([9, 8, 7])
     >>> a
     array([ 9.+2.j,  8.+4.j,  7.+6.j])
+    >>> np.real(1 + 1j)
+    1.0
 
     """
-    return asanyarray(val).real
+    arr = asanyarray(val)
+    return arr.real.item() if _nx.isscalar(val) else arr.real
```

Comment:
Just thinking about duct typing here... How about:

```python
try:
    return val.real
except AttributeError:
    return asanyarray(val).real
```

Which has the nice bonus that builtin `complex` returns builtin `float` not `np.float64`